typescript/web/src/components/export-button/export-modal/export-options-modal.tsx

```diff
@@ -100,12 +103,17 @@ export const ExportOptionsModal = () => {
     [exportFormatLowerCase]
   );
 
-  const handleClick = useCallback(() => {
+  const exportAction = useCallback(() => {
     exportFunction(exportOptions);
     trackEvent(`export_button_click_${exportFormat.toLocaleLowerCase()}`, {});
     setIsOptionsModalOpen(false);
   }, [exportFormat, exportFunction, exportOptions, setIsOptionsModalOpen]);
 
+  useEffect(() => {
```

Comment:
Why do you need a `useEffect` exactly? Can't it be avoided?

Reply:
I'm not sure, but if I don't use it I get this in the console when clicking on the CSV export card:

```
Warning: Cannot update a component (`ExportModalProvider`) while rendering a different component (`ExportOptionsModal`).
```

I'm no expert in React hooks, I would appreciate your help on that.